Fixed migrations so you can rollback them all

aiperon opened the pull request:
## Summary

This PR resolves issues with errors on rollback migrations. Specifically, it:
- Fixes an incorrect migration method name in AddApplyToAllToVariantPropertyRule.
- Fixes an incorrect foreign key removal. 

Additionally, it modifies the first migration to drop all Spree tables in its down method. This ensures that the database remains clean after a full rollback.

## Checklist

Check out our [PR guidelines](https://github.com/solidusio/.github/blob/master/CONTRIBUTING.md#pull-request-guidelines) for more details.

The following are mandatory for all PRs:

- [x] [I agree that my PR will be published under the same license as Solidus](https://github.com/solidusio/solidus/blob/main/LICENSE.md).
- [x] I have written a thorough PR description.
- [x] I have kept my commits small and atomic.
- [x] [I have used clear, explanatory commit messages](https://github.com/solidusio/.github/blob/main/CONTRIBUTING.md#writing-good-commit-messages).

jarednorman reviewed (approved):
Don't know why you'd need to roll back migrations this far back, but seems fine to me.

aiperon commented:
> Don't know why you'd need to roll back migrations this far back, but seems fine to me.

I added Solidus to the existing app and experienced some issues due to my installation choices. When I reverted it back to the initial state, I found these issues. I wanted to preserve my existing database before another attempt.

tvdeyen reviewed (approved):
Nice. Thanks
